### ISSUE TITLE
Sync issue overwrote my note

### ISSUE THREAD

**Issue by nth-chile** [glushchenko/fsnotes#1251]
### Description

A syncing error between iOS and mac app caused the latest version of my note to get lost

### To Reproduce

This is what I did, but I'm sure it's not consistently reproducable.

1. Edit a note on mac app, close computer
2. Later, open the note with iOS app. The note is not up to date
3. Edit
4. Later, open the note with mac app. It reflects the phone version, and what you wrote in step 1 has been lost

### Expected behavior

Any way to ensure that changes will never be lost

### FSNotes version

5.1.3

### macOS/iOS version

4.8.5

**Comment by gingerbeardman** [comment]
Sync is managed by iCloud, it should create a second file when there is a conflict. So there would be two files and you'd have to resolve the clash yourself.

**Comment by nth-chile** [comment]
I've seen that before, but this time it didn't happen

**Comment by nth-chile** [comment]
Out of curiosity, when you say that sync is managed by iCloud, do you mean that iCloud creates the conflict file?

**Comment by gingerbeardman** [comment]
FSNotes created the second file.

https://github.com/glushchenko/fsnotes/blob/d113f9e0aa36b44fd66362ed4df38e539d1c3170/FSNotes%20iOS/CloudDriveManager.swift#L312-L350

But iCloud reports the conflict: https://developer.apple.com/documentation/foundation/nsfileversion/1417854-unresolvedconflictversionsofitem

```
        if let conflicts = NSFileVersion.unresolvedConflictVersionsOfItem(at: url as URL) {
```

**Comment by nth-chile** [comment]
That's cool/good to know. We can close this since it's not easily reproducible and because it might be out of our hands, unless you guys think there is a possibility that  the FSNotes code is to blame.

@glushchenko

**Comment by loong1992** [comment]
When the notes are encrypted, the BUG will reproduce. This BUG is not an iCloud BUG.
1. Both the computer and the phone unlock the encrypted notes.
2. Then the iOS app edits the notes, by which time iCloud has synced the encrypted files to the computer.
3. However, the Mac app will not reload the updated files, this time still display the old files, and then use the computer editing will lose the contents of the phone changes.

**Comment by nth-chile** [comment]
Today, this happened: I updated a note on my mac. Left the house, opened the iOS app which was already running in the background with the note open. I saw the old version of the note appear immediately but it didn't update. I knew that editing it might cause the new version to disappear, so I just quit the app (double tap home button, swipe up to quit) and relaunched and the new note was there.

**Comment by glushchenko** [comment]
Hi, guys!

I made some improvements for sync conflicts, no more overwrites for encrypted notes. I think bugs was resolved in FSNotes 5.2.0.

https://github.com/glushchenko/fsnotes/releases/download/5.2.0/FSNotes_5.2.0.zip
